ign_rviz_plugins/include/ignition/rviz/plugins/GlobalOptions.hpp

```diff
@@ -56,6 +56,33 @@ class GlobalOptions : public MessageDisplayBase
     NOTIFY frameListChanged
   )
 
+  /**
+   *  @brief TF status
+   */
+  Q_PROPERTY(
+    QString tfStatus
+    READ getTfStatus
+    NOTIFY tfStatusChanged
+  )
+
+  /**
+   *  @brief TF status message
+   */
+  Q_PROPERTY(
+    QString tfStatusMessage
+    READ getTfStatusMessage
+    NOTIFY tfStatusMessageChanged
+  )
+
+  /**
+   *  @brief TF status color
+   */
+  Q_PROPERTY(
+    QString tfStatusColor
+    READ getTfStatusColor
+    NOTIFY tfStatusColorChanged
+  )
```

Comment:
I believe instead of having 3 properties to read and notify the changes, we can have a custom `Q_OBJECT` class with these properties i.e status, message, and color along with methods to set these properties which are used in [`updateTfStatus`](https://github.com/ignitionrobotics/ign-rviz/blob/69070895603eecf2d799d15ec5a030ef30450378/ign_rviz_plugins/src/rviz/plugins/GlobalOptions.cpp#L80-L114).

This class can then be extended in the future to implement the tree view with the TF status of each frame.